@afishbeck I am also getting a problem with:

/usr/bin/ld: /home/gavin/dev/hpcc/testing/unittests/loggingtests.cpp:166: undefined reference to `createGetTransactionSeedResponse'

which is missing a DECL_EXPORT (I'm not sure why)

--- a/configuration/configmgr/hpcc_support/add_required_instances/AddRequiredInstancesSupport.cpp
+++ b/configuration/configmgr/hpcc_support/add_required_instances/AddRequiredInstancesSupport.cpp
@@ -18,8 +18,7 @@
 #include <algorithm>
 #include "AddRequiredInstancesSupport.hpp"
 
-
-extern "C" HPCCConfigSupport* getCfgMgrSupportInstance(EnvironmentMgr *pEnvMgr)
+extern "C" DECL_EXPORT HPCCConfigSupport* getCfgMgrSupportInstance(EnvironmentMgr *pEnvMgr)
 {
     HPCCConfigSupport *pInstance = new AddRequiredInstancesSupport(pEnvMgr);
     return pInstance;

--- a/configuration/configmgr/hpcc_support/check_port_bindings/CheckPortBindings.cpp
+++ b/configuration/configmgr/hpcc_support/check_port_bindings/CheckPortBindings.cpp
@@ -24,7 +24,7 @@
 #include <algorithm>
 
 
-extern "C" HPCCConfigSupport* getCfgMgrSupportInstance(EnvironmentMgr *pEnvMgr)
+extern "C" DECL_EXPORT HPCCConfigSupport* getCfgMgrSupportInstance(EnvironmentMgr *pEnvMgr)
 {
     HPCCConfigSupport *pInstance = new CheckPortBindings(pEnvMgr);
     return pInstance;

--- a/testing/unittests/CMakeLists.txt
+++ b/testing/unittests/CMakeLists.txt
@@ -30,6 +30,8 @@ endif()
 
 project( unittests )
 
+include(${HPCC_SOURCE_DIR}/esp/scm/espscm.cmake)
+
 set (    SRCS
          unittests.cpp
          remotetests.cpp
@@ -42,8 +44,9 @@ set (    SRCS
          txSummarytests.cpp
          accessmaptests.cpp
          fxpptests.cpp
-         ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp/fxpp/FragmentedXmlPullParser.cpp;
-         ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp/fxpp/FragmentedXmlAssistant.cpp;
+         ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp/fxpp/FragmentedXmlPullParser.cpp
+         ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp/fxpp/FragmentedXmlAssistant.cpp
+         ${CMAKE_BINARY_DIR}/generated/ws_loggingservice_esp.cpp
     )
 
 if (NOT CONTAINERIZED)

